### ISSUE TITLE
getEntityRecord issues a HTTP request when there's a cached result

### ISSUE THREAD

**Issue by adamziel** [WordPress/gutenberg#24308]
**Describe the bug**

Consider this fragment:

https://github.com/WordPress/gutenberg/blob/be014202a75449c0afc384f71835c04a048e53e2/packages/core-data/src/resolvers.js#L57-L59

I would expect it to short-circuit if the entity is found, but it does the opposite which results in extra http queries.

For example, doing:
```js
getEntityRecords("root", "widgetArea");
// ... wait for resolution ...
getEntityRecord("root", "widgetArea", "sidebar-1");
```
Will issue two http queries even if the entity with id `sidebar-1` is already available.

CC @youknowriad

**Comment by youknowriad** [comment]
I wonder if this is solved with #19498 which is close to land.

**Comment by adamziel** [comment]
> I wonder if this is solved with #19498 which is close to land.

It seems like it will! Let's confirm once it lands and then hopefully close this issue.

Also, in reply to https://github.com/WordPress/gutenberg/pull/24290#discussion_r466328591:

> this change is problematic for me. entity will always be defined aside on the very first call so getEntityRecord won't do anything for all calls except the first one?

I may be missing something, but it would make sense to me to use the entity we already have (until the cache is invalidated) instead of re-requesting a new one each time. I don't get how it's short-circuiting when the entity is not found.

**Comment by youknowriad** [comment]
> I may be missing something, but it would make sense to me to use the entity we already have (until the cache is invalidated) instead of re-requesting a new one each time. I don't get how it's short-circuiting when the entity is not found - shouldn't that be exactly when the resolution needs to happen?.

I think you're confusing "Entity" and "Entity Record". "Entity" is just the config of the entity, we're resolving the record at this point.

**Comment by adamziel** [comment]
Ah I see, it all makes sense now! Awesome, so #19498 will solve this.
